app/models/concerns/foreman_remote_execution/orchestration/ssh.rb

```diff
@@ -0,0 +1,65 @@
+module ForemanRemoteExecution
+  module Orchestration::SSH
+    extend ActiveSupport::Concern
+
+    included do
+      before_destroy :ssh_destroy
+      after_validation :queue_ssh_destroy
+    end
+
+    def drop_from_known_hosts
+      host, target = if self.is_a?(::Host::Base)
+                       [self, name]
+                     else
+                       [self.host, ip]
+                     end
+      proxies = %w(SSH Ansible).map { |kind| host.remote_execution_proxies(kind).values }
+      proxies.flatten.uniq.each do |proxy|
+        begin
+          proxy.drop_host_from_known_hosts(target)
+        rescue => e
+          Rails.logger.warn e.message
+        end
+      end
+    end
+
+    def ssh_destroy
+      logger.debug "Scheduling SSH known_hosts cleanup"
+      queue.create(id: queue_id, name: _("Remove SSH known hosts for %s") % self,
+        priority: 200, action: [self, :drop_from_known_hosts])
+    end
+
+    def queue_ssh_destroy
+      return unless should_drop_from_known_hosts?
+      ssh_destroy
+    end
+
+    def should_drop_from_known_hosts?
+      host = self.is_a?(::Host::Base) ? self : self.host
+      host.build && host.changes.key?('build')
+    end
+
+    private
+
+    def queue_id
+      kind, id = if self.is_a?(::Host::Base)
+                   ['host', name]
+                 else
+                   ['interface', ip]
+                 end
+      "ssh_remove_known_hosts_#{kind}_#{id}"
+    end
+  end
+
+  module Orchestration::SSHManaged
+    extend ActiveSupport::Concern
+
+    included do
+      register_rebuild(:rebuild_ssh, N_("SSH_#{self.to_s.split('::').first}"))
```

Comment:
What is this good for? I wasn't able to trigger anything using just this, I had to trigger things from after_validation callback